tests/src/usecases/budget.spec.ts

```diff
@@ -0,0 +1,481 @@
+			const input = {
+				accountId: 'exemploAccountId',
+				budgetId: '1',
+				month: 2,
+				year: 2024,
+			};
+
+			categoryRepository.mock.getByUser.mockResolvedValue(
+				categoryRepository.outputs.getByUser.inactiveCategory,
+			);
+
+			let result;
+			try {
+				result = await service.overview(input);
+			} catch (err) {
+				result = err;
+			}
+
+			expect(categoryRepository.mock.getByUser).toHaveBeenCalledWith({
+				accountId: input.accountId,
+				limit: 10000,
+				offset: 0,
+			});
+			expect(budgetRepository.mock.getMonthlyByCategory).toHaveBeenCalledWith(
+				input,
+			);
+			expect(
+				transactionRepository.mock.getMonthlyAmountByCategory,
+			).toHaveBeenCalledWith(input);
+			expect(result.totalBudget).toEqual(450);
+			expect(result.totalExpenses).toEqual(150);
+			expect(result.remainingBudget).toEqual(300);
+			expect(result.budgetByCategory).toHaveLength(2);
+			expect(result.budgetByCategory).toEqual([
+				{
+					id: 1,
+					name: 'Category A',
+					active: false,
+					totalExpenses: 50,
+					totalBudget: 100,
+					remainingBudget: 50,
+					color: 'red',
+					description: 'description',
+					icon: 'bank',
+				},
+				{
+					id: 3,
+					name: 'Category C',
+					active: false,
+					totalExpenses: 100,
+					totalBudget: 150,
+					remainingBudget: 50,
+					color: 'red',
+					description: 'description',
+					icon: 'bank',
+				},
+			]);
+		});
+		it('should correctly handle negative budgets and expenses', async () => {
+			const input = {
+				accountId: 'exemploAccountId',
+				budgetId: '1',
+				month: 2,
+				year: 2024,
+			};
+
+			budgetRepository.mock.getMonthlyByCategory.mockResolvedValue(
+				budgetRepository.outputs.getMonthlyByCategory.negativeBudget,
+			);
+			transactionRepository.mock.getMonthlyAmountByCategory.mockResolvedValue(
+				transactionRepository.outputs.getMonthlyAmountByCategory
+					.expenseNegative,
+			);
+
+			let result;
+			try {
+				result = await service.overview(input);
+			} catch (err) {
+				result = err;
+			}
+
+			expect(categoryRepository.mock.getByUser).toHaveBeenCalledWith({
+				accountId: input.accountId,
+				limit: 10000,
+				offset: 0,
+			});
+			expect(budgetRepository.mock.getMonthlyByCategory).toHaveBeenCalledWith(
+				input,
+			);
+			expect(
+				transactionRepository.mock.getMonthlyAmountByCategory,
+			).toHaveBeenCalledWith(input);
+
+			expect(result.totalBudget).toEqual(120);
+			expect(result.totalExpenses).toEqual(250);
+			expect(result.remainingBudget).toEqual(-130);
+			expect(result.budgetByCategory).toHaveLength(3);
+		});
+	});
+
+	describe('> createNextBudgetDates', () => {
+		it('should generates future budget dates for a specific budget', async () => {
+			const input = {
+				startFrom: {
+					id: '1',
+					budgetId: '1',
+					month: 2,
+					year: 2024,
+					date: new Date(2024, 3, 2),
+				},
+				amount: 20,
+			};
+			const dates = Array.from({ length: input.amount }, (_, index) => {
```

Comment:
Use raw values or simple logic instead of a bit more complex logic, or else we would have to test it too.